source/slang/slang-ir-ssa-simplification.h

```diff
@@ -5,13 +5,14 @@ namespace Slang
 {
     struct IRModule;
     struct IRGlobalValueWithCode;
+    class DiagnosticSink;
 
     // Run a combination of SSA, SCCP, SimplifyCFG, and DeadCodeElimination pass
     // until no more changes are possible.
-    void simplifyIR(IRModule* module);
+    void simplifyIR(IRModule* module, DiagnosticSink* sink = nullptr);
```

Comment:
The sink seems to be the mechanism to report the error to caller. Should we require a sink to be passed?

Reply:
There are several calls to `simplifyIR` from a pass that currently don't have access to `sink`. Plumbing `sink` through every pass feels messy. We should probably have a base context type for a pass that carries the sink over to simplify things. But for now this is fine since all errors will eventually be captured by a `simplifyIR` call in `linkAndOptimizeIR`, where a sink is provided there. When `sink` is not provided, the compilation can still continue normally upon a div-by-zero error (we simply don't fold).